### ISSUE TITLE
File descriptors leak causes system to crash

### ISSUE THREAD

**Issue by ducarroz** [Microsoft/vscode#28750]
- VSCode Version: Code 1.13.0 (376c52b955428d205459bea6619fc161fc8faacf, 2017-06-08T16:34:53.678Z)
- OS Version: Darwin x64 16.6.0
- Extensions: 

|Extension|Author|Version|
|---|---|---|
|jshint|dbaeumer|0.10.15|
|vscode-eslint|dbaeumer|1.2.11|
|githistory|donjayamanne|0.2.0|
|debugger-for-chrome|msjsdiag|3.1.3|
|darcula-extended|smlombardi|3.3.2|
|vscode-open-in-github|sysoev|1.6.0|
|gitblame|waderyan|1.11.2|
|Material-theme|zhuangtongfa|2.8.6|;

---
I have a project with a lot of npm dependencies. I also run an ember-cli server and a react server. After couple minutes of usage, my OS ran out of file descriptors and nothing works anymore. When I look at the number of open files per process, VSCode is always at the top of the list and shows 8000+ open files. As soon I quit VSCode, everything is back to normal.

This is not something progressive where I would see the number of open file going up slowly. VSCode is pretty stable until I am suddenly put of file descriptors.

I haven't found yet what steps causes it but it occurs all the time! That issue started to show up in the past week or two.

**Comment by ducarroz** [comment]
looking at the list of open files. Looks like VSCode opens all the files in <project>/tmp generated by ember-cli!

**Comment by mjbvz** [comment]
Can you try disabling your extensions to see if one of those is the root cause. Just start code with the `--disable-extensions` command line flag

**Comment by ducarroz** [comment]
I am not able to reproduce it today, will keep an eye on it...

**Comment by mjbvz** [comment]
Ok, please let me know if you see this issue again with extensions disabled

**Comment by lyahdav** [comment]
@mjbvz I just had this happen to me when opening VSCode on a large Android project. I tried launching with `--disable-extensions` but it didn't help. My machine had so many open files that it got really unstable. After quitting VSCode it resolved. `lsof -d "^txt" | wc -l` returned 11945. I also verified the majority of those files were from VSCode (7983 at one point).

Should we reopen this issue or should I create a new one?
